controllers/designate_controller.go

```diff
@@ -574,6 +579,17 @@ func (r *DesignateReconciler) reconcileNormal(ctx context.Context, instance *des
 	}
 	// Create ConfigMaps and Secrets - end
 
+	// ensure MariaDBAccount exists.  This account record may be created by
+	// openstack-operator or the cloud operator up front without a specific
+	// MariaDBDatabase configured yet.   Otherwise, a MariaDBAccount CR is
+	// created here with a generated username as well as a secret with
+	// generated password.   The MariaDBAccount is created without being
+	// yet associated with any MariaDBDatabase.
+	_, _, err = mariadbv1.EnsureMariaDBAccount(
+		ctx, helper, instance.Spec.DatabaseAccount,
+		instance.Namespace, false,
+	)
+
```

Comment:
need to add a status update here, and also return if err is non nil, example https://github.com/openstack-k8s-operators/neutron-operator/blame/13c5c990d917ab2825bda8e44b6f631474cc9fe1/controllers/neutronapi_controller.go#L925